catalog/resource_sql_table.go

```diff
@@ -0,0 +1,259 @@
+		notNull = " NOT NULL"
+	}
+
+	comment := ""
+	if col.Comment != "" {
+		comment = fmt.Sprintf(" COMMENT %s", col.Comment)
+	}
+	return fmt.Sprintf("%s %s%s%s", col.Name, col.TypeText, notNull, comment) // id INT NOT NULL COMMENT something
+}
+
+func (ti *SqlTableInfo) serializeColumnInfos() string {
+	columnFragments := make([]string, len(ti.ColumnInfos))
+	for i, col := range ti.ColumnInfos {
+		columnFragments[i] = ti.serializeColumnInfo(col)
+	}
+	return strings.Join(columnFragments[:], ", ") // id INT NOT NULL, name STRING, age INT
+}
+
+func (ti *SqlTableInfo) serializeProperties() string {
+	propsMap := make([]string, 0, len(ti.Properties))
+	for key, value := range ti.Properties {
+		propsMap = append(propsMap, fmt.Sprintf("'%s'='%s'", key, value))
+	}
+	return strings.Join(propsMap[:], ", ") // 'foo'='bar', 'this'='that'
+}
+
+func (ti *SqlTableInfo) buildTableCreateStatement() string {
+	statements := make([]string, 0, 10)
+
+	isView := ti.TableType == "VIEW"
+
+	externalFragment := ""
+	if ti.TableType == "EXTERNAL" {
+		externalFragment = "EXTERNAL "
+	}
+
+	createType := "TABLE"
+	if isView {
+		createType = "VIEW"
+	}
+
+	statements = append(statements, fmt.Sprintf("CREATE %s%s %s", externalFragment, createType, ti.FullName()))
+
+	if len(ti.ColumnInfos) > 0 {
+		statements = append(statements, fmt.Sprintf(" (%s)", ti.serializeColumnInfos()))
+	}
+
+	if ti.Comment != "" {
+		statements = append(statements, fmt.Sprintf("\nCOMMENT '%s'", ti.Comment)) // COMMENT 'this is a comment'
+	}
+
+	if len(ti.Properties) > 0 {
+		statements = append(statements, fmt.Sprintf("\nTBLPROPERTIES (%s)", ti.serializeProperties())) // TBLPROPERTIES ('foo'='bar')
+	}
+
+	if !isView {
+		if ti.DataSourceFormat != "" {
+			statements = append(statements, fmt.Sprintf("\nUSING %s", ti.DataSourceFormat)) // USING CSV
+		}
+
+		if ti.StorageLocation != "" {
+			statements = append(statements, fmt.Sprintf("\nLOCATION '%s'", ti.StorageLocation)) // LOCATION '/mnt/csv_files'
+
+			if ti.StorageCredentialName != "" {
+				statements = append(statements, fmt.Sprintf(" WITH (CREDENTIAL `%s`)", ti.StorageCredentialName))
+			}
+		}
+	} else {
+		statements = append(statements, fmt.Sprintf("\nAS %s", ti.ViewDefinition))
+	}
+
+	statements = append(statements, ";")
+
+	return strings.Join(statements, "")
+}
+
+func (ti *SqlTableInfo) createTable(ctx context.Context, d *schema.ResourceData, s map[string]*schema.Schema, c *common.DatabricksClient) error {
+
+	sqlQuery := ti.buildTableCreateStatement()
+
+	log.Printf("[INFO] Executing Sql: %s", sqlQuery)
+	r := ti.exec.Execute(ti.ClusterID, "sql", sqlQuery)
+
+	if !r.Failed() {
+		return nil
+	}
+	return fmt.Errorf("cannot execute %s: %s", sqlQuery, r.Error())
+}
+
+func ResourceSqlTable() *schema.Resource {
+	tableSchema := common.StructToSchema(SqlTableInfo{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			return m
+		})
+	update := updateFunctionFactory("/unity-catalog/tables", []string{
```

Comment:
the Unity Catalog team told us that update API is also internal, so update calls should also be done via SQL warehouse/cluster

Reply:
Done